### ISSUE TITLE
Can we store IDP of user in the FileSender DB then filter configs on it?

### ISSUE THREAD

**Issue by madsi1m** [filesender/filesender#2081]
Hi, I have been exploring `$config['auth_config_regex_files']` and figure it would be much cleaner and easier if we could just filter on IDPs coming from SimpleSAMLPHP rather than having to learn each IDPs list of email domains.

Looking at `attributes()` function in `classes/auth/AuthSPSaml.class.php` we can query the IDP from SimpleSAMLPHP by doing:
```
$attributes['idp'] = parse_url(trim($ssp->getAuthData('saml:sp:IdP')), PHP_URL_HOST);
Logger::info('User IDP: '.$attributes['idp']);
```

Could we add this as a field in the `Authentications` table? (or similar) This will help us generate graphs per idp which much simpler SQLs.

And could we extend `$config['auth_config_regex_files']` to have an `idp` filter? For example:
```
  $config['auth_config_regex_files'] = [
      'uid' => [
          '@mydomain.com$' => 'mydomainfile',
//          '@myotherdomain.com$|@yetanotherdomain.com$' => 'myotherdomainfile',  //now using IDP filter below
      ],
      'idp' => [
          'idp.myotherdomain.com$' => 'myotherdomainfile',
          'idp.institute2.com$' => 'institute2',
    ],
  ];
```

**Comment by madsi1m** [comment]
edited above comment to also put idp in `$attributes` which as far as i can see will make `$config['auth_config_regex_files']` work

**Comment by WebSpider** [comment]
Isnt this what `auth_sp_additional_attributes` and `auth_sp_save_user_additional_attributes` is for?

**Comment by madsi1m** [comment]
not exactly, as the IDP name/host is not actually an attribute that is returned with user information like "sn" or "department". The info actually comes from the `saml:SP` API

**Comment by madsi1m** [comment]
>Can we store IDP of user in the FileSender DB then filter configs on it?
Seems Yes: https://github.com/filesender/filesender/pull/2082

**Comment by monkeyiq** [comment]
I guess the fix merged label is to early. The IdP stuff is there but the config selection isn't yet merged ;)
